### PR TITLE
Use the XML reporting API

### PR DESCRIPTION
**Pull request by mulby**
@brianhw @jab5569 this is the paypal discovery work

after much tweaking it appears to produce the same data as the previous method but allows us to pull it incrementally

I don't really like how this code is structured, so I'm treating it as effectively throw-away, but it does prove out the strategy

### CODE REVIEW

**Comment by mulby** [comment]
Also - I checked one day worth of data against the version currently produced by the financial-reporting branch and it matched up exactly

File: edx/analytics/tasks/reports/paypal.py [hunk 1]
```diff
@@ -18,183 +22,436 @@
...
+
+    @classmethod
+    def params_from_xml(cls, root_node):
+        base_response_node = root_node.find('baseResponse')
+        return {
+            'response_code': int(base_response_node.findtext('responseCode')),
+            'response_message': base_response_node.findtext('responseMsg')
         }
 
-        log.debug('Requesting payments')
```
Comment (johnalbaker) [inline edx/analytics/tasks/reports/paypal.py:69]: Nit: Remove

**Comment by johnalbaker** [comment]
:+1: LGTM

**Comment by brianhw** [comment]
LGTM.   Some things to look at:  
- There is no timeout on the report polling, which makes me nervous.  If for some reason their report generation gets stuck, then our report retrieval also gets stuck.
- It's pretty elegant to have each response try to find the appropriate class to parse the response, and then instantiate that class based on the params that the class parsed.  But it also seems a little twisted, since we presumably know what kind of response we're expecting, so we should be able to just instantiate it directly by passing the response class name into the send method. 
- I wasn't clear on the decimal processing -- do they really return integers with number of cents?  That seems really weird if the currency is dollars.

**Comment by mulby** [comment]
@brianhw - a few responses to your points
- Definitely... in general the exception handling and non-happy path cases need much more careful review.
- Yeah that was part of the reason why I wanted to refactor this code. There is a very clear request-response protocol that this code doesn't do a good job of communicating / leveraging.
- Yes - they return integer number of cents. I agree this is unintuitive.

**Pull request closed without merging**